contracts/token/ERC721/ERC721.sol

```diff
@@ -331,8 +331,24 @@ contract ERC721 is Context, ERC165, IERC721 {
             return true;
         }
 
-        bytes4 retval = IERC721Receiver(to).onERC721Received(_msgSender(), from, tokenId, _data);
-        return (retval == _ERC721_RECEIVED);
+        bytes memory payload = abi.encodeWithSignature(
+            "onERC721Received(address,address,uint256,bytes)",
```

Comment:
Please use `abi.encodeWithSelector` with `IERC721Receiver(to).onERC721Received.selector` as the first argument. This ensures that we don't mess up the string constant.